### ISSUE TITLE
Launcher crashes every time on Fedora 34 Workstation x86_64

### ISSUE THREAD

**Issue by lorecast162** [AM2R-Community-Developers/AM2RLauncher#12]
I'm trying to launch am2r on my laptop running F34 but it crashes constantly.
To install dotnet5 I installed the `aspnetcore-runtime-5.0` package which should work as it still does contain dotnet5 core runtime stuff.
To install libappindicator3 I tried with `libappindicator` first but it complained about not finding `liblibappindicator3.so.1` so I retried with `libappindicator-gtk3` which got me further, but not much as you'll see later.
For xdelta I installed `xdelta`.
Upon running this is what I get:
![image](https://user-images.githubusercontent.com/17319563/119239695-070a7800-bb4b-11eb-90f1-b3645d722044.png)

This then goes on until I see part of the interface which is still stuck in that crash loop (also follows log):
```
2021-05-22 22:10:16,147 [1] INFO AM2RLauncher.MainForm - Running update check...
2021-05-22 22:10:16,222 [1] INFO AM2RLauncher.MainForm - AutoUpdate set to true!
2021-05-22 22:10:17,455 [1] INFO AM2RLauncher.MainForm - Checking internet connection...
2021-05-22 22:10:17,649 [1] INFO AM2RLauncher.MainForm - Internet connection established!
2021-05-22 22:10:17,649 [1] INFO AM2RLauncher.MainForm - Mutex check passed. Entering main thread.
2021-05-22 22:10:17,649 [1] INFO AM2RLauncher.MainForm - Current Launcher Version: 2.0.0
2021-05-22 22:10:17,650 [1] INFO AM2RLauncher.MainForm - Current Platform-ID is: Gtk
2021-05-22 22:10:17,652 [1] INFO AM2RLauncher.MainForm - Language has been set to: English (United States)
2021-05-22 22:10:17,652 [1] INFO AM2RLauncher.MainForm - Beginning UI initialization...
2021-05-22 22:10:17,797 [1] INFO AM2RLauncher.MainForm - Randomly chosen splash: What you are referring to as AM2R, is in fact, GNU/AM2R.
2021-05-22 22:10:17,934 [1] INFO AM2RLauncher.MainForm - Start the launcher with Size: 0, 0
2021-05-22 22:10:18,697 [1] INFO AM2RLauncher.MainForm - All UI objects have been initalized, UI has been set up.
2021-05-22 22:10:18,697 [1] INFO AM2RLauncher.MainForm - Beginning event linkage...
2021-05-22 22:10:18,700 [1] INFO AM2RLauncher.MainForm - Events linked successfully.
2021-05-22 22:10:19,130 [1] ERROR AM2RLauncher.MainForm - An unhandled exception has occurred: 
*****Stack Trace*****

System.NullReferenceException: Object reference not set to an instance of an object.
   at Gtk.Widget.DrawnSignalCallback(IntPtr inst, IntPtr arg0, IntPtr gch)
2021-05-22 22:10:22,247 [1] ERROR AM2RLauncher.MainForm - An unhandled exception has occurred: 
*****Stack Trace*****

System.NullReferenceException: Object reference not set to an instance of an object.
   at Gtk.Widget.DrawnSignalCallback(IntPtr inst, IntPtr arg0, IntPtr gch)
2021-05-22 22:10:22,987 [1] ERROR AM2RLauncher.MainForm - An unhandled exception has occurred: 
*****Stack Trace*****

System.NullReferenceException: Object reference not set to an instance of an object.
   at Gtk.Widget.DrawnSignalCallback(IntPtr inst, IntPtr arg0, IntPtr gch)
2021-05-22 22:10:23,514 [1] ERROR AM2RLauncher.MainForm - An unhandled exception has occurred: 
*****Stack Trace*****

System.NullReferenceException: Object reference not set to an instance of an object.
   at Gtk.Widget.DrawnSignalCallback(IntPtr inst, IntPtr arg0, IntPtr gch)
2021-05-22 22:10:24,008 [1] ERROR AM2RLauncher.MainForm - An unhandled exception has occurred: 
*****Stack Trace*****

System.NullReferenceException: Object reference not set to an instance of an object.
   at Gtk.Widget.DrawnSignalCallback(IntPtr inst, IntPtr arg0, IntPtr gch)
2021-05-22 22:10:24,538 [1] ERROR AM2RLauncher.MainForm - An unhandled exception has occurred: 
*****Stack Trace*****

System.NullReferenceException: Object reference not set to an instance of an object.
   at Gtk.Widget.DrawnSignalCallback(IntPtr inst, IntPtr arg0, IntPtr gch)
2021-05-22 22:10:24,990 [1] ERROR AM2RLauncher.MainForm - An unhandled exception has occurred: 
*****Stack Trace*****

System.NullReferenceException: Object reference not set to an instance of an object.
   at Gtk.Widget.DrawnSignalCallback(IntPtr inst, IntPtr arg0, IntPtr gch)
2021-05-22 22:10:25,604 [1] ERROR AM2RLauncher.MainForm - An unhandled exception has occurred: 
*****Stack Trace*****

System.NullReferenceException: Object reference not set to an instance of an object.
   at Gtk.Widget.DrawnSignalCallback(IntPtr inst, IntPtr arg0, IntPtr gch)
2021-05-22 22:10:26,321 [1] ERROR AM2RLauncher.MainForm - An unhandled exception has occurred: 
*****Stack Trace*****

System.NullReferenceException: Object reference not set to an instance of an object.
   at Gtk.Widget.DrawnSignalCallback(IntPtr inst, IntPtr arg0, IntPtr gch)
2021-05-22 22:10:27,333 [1] ERROR AM2RLauncher.MainForm - An unhandled exception has occurred: 
*****Stack Trace*****

System.NullReferenceException: Object reference not set to an instance of an object.
   at Gtk.Widget.DrawnSignalCallback(IntPtr inst, IntPtr arg0, IntPtr gch)
2021-05-22 22:10:27,958 [1] ERROR AM2RLauncher.MainForm - An unhandled exception has occurred: 
*****Stack Trace*****

System.NullReferenceException: Object reference not set to an instance of an object.
   at Gtk.Widget.DrawnSignalCallback(IntPtr inst, IntPtr arg0, IntPtr gch)
2021-05-22 22:10:28,717 [1] ERROR AM2RLauncher.MainForm - An unhandled exception has occurred: 
*****Stack Trace*****

System.NullReferenceException: Object reference not set to an instance of an object.
   at Gtk.Widget.DrawnSignalCallback(IntPtr inst, IntPtr arg0, IntPtr gch)
2021-05-22 22:10:29,332 [1] ERROR AM2RLauncher.MainForm - An unhandled exception has occurred: 
*****Stack Trace*****

System.NullReferenceException: Object reference not set to an instance of an object.
   at Gtk.Widget.DrawnSignalCallback(IntPtr inst, IntPtr arg0, IntPtr gch)
2021-05-22 22:10:29,623 [1] ERROR AM2RLauncher.MainForm - An unhandled exception has occurred: 
*****Stack Trace*****

System.NullReferenceException: Object reference not set to an instance of an object.
   at Gtk.Widget.DrawnSignalCallback(IntPtr inst, IntPtr arg0, IntPtr gch)
2021-05-22 22:10:29,862 [1] ERROR AM2RLauncher.MainForm - An unhandled exception has occurred: 
*****Stack Trace*****

System.NullReferenceException: Object reference not set to an instance of an object.
   at Gtk.Widget.DrawnSignalCallback(IntPtr inst, IntPtr arg0, IntPtr gch)
```
That little error box continues showing up after I press OK until I see this in my terminal:
```
Fatal error. System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
   at Gtk.Widget.DrawnSignalCallback(IntPtr, IntPtr, IntPtr)
   at Eto.GtkSharp.Forms.ApplicationHandler.Run()
   at Eto.Forms.Application.Run(Eto.Forms.Form)
   at AM2RLauncher.Gtk.MainClass.Main(System.String[])
[1]    4255 IOT instruction (core dumped)  ./AM2RLauncher.Gtk
```

I haven't tried compiling yet but I will after posting this. Will keep you updated.

**Comment by Miepee** [comment]
@lorecast162 What GlibC version do you have?

**Comment by lorecast162** [comment]
@Miepee I used `dnf info glibc` and I got version 2.33.
I just compiled AM2RLauncher and it seems to work just fine.
Just the launcher tho, still haven't finished the "Download" part in the launcher

**Comment by lorecast162** [comment]
It's been stuck here for a while (at least from when I posted the last reply) but I don't see any errors.
![image](https://user-images.githubusercontent.com/17319563/119240106-00c9cb00-bb4e-11eb-92d7-43b1304d4590.png)

**Comment by Lojemiru** [comment]
The Launcher crash is a known issue with Eto.forms, it's been resolved but we have yet to make a release to push the fix. Getting a few things sorted out first.

**Comment by Miepee** [comment]
Should be fixed as per release 2.1.0
